app/controllers/admin/facilities_controller.rb

```diff
@@ -5,44 +5,35 @@ class Admin::FacilitiesController < AdminController
 
   before_action :set_facility, only: [:show, :edit, :update, :destroy]
   before_action :set_facility_group, only: [:show, :new, :create, :edit, :update, :destroy]
+  before_action :authorize_facility, only: [:show, :new, :create, :edit, :update, :destroy]
   before_action :initialize_upload, :validate_file_type, :validate_file_size, :parse_file,
     :validate_facility_rows, if: :file_exists?, only: [:upload]
 
   def index
-    authorize([:manage, :facility, Facility])
+    authorize([:upcoming, :manage, Facility], :allowed?)
 
-    if searching?
-      facilities = policy_scope([:manage, :facility, Facility]).search_by_name(search_query)
-      facility_groups = FacilityGroup.where(facilities: facilities)
+    facilities =
+      if searching?
+        policy_scope([:upcoming, :manage, Facility]).search_by_name(search_query)
```

Comment:
Suggestion:

```
current_admin.facilities.search_by_name(query)
```

Reply:
Again, I'm not certain what `current_admin.facilities` means. It's not clear looking at [Take 3](https://github.com/simpledotorg/simple-server/pull/1173) how this would work.

Refer to [Take 2](https://github.com/simpledotorg/simple-server/pull/1160/files#diff-dbee6fa62dba5eb76fea1e205e2a5f54R52) to see a nicer implementation for this.